zio-http/src/main/scala/zhttp/http/URL.scala

```diff
@@ -45,12 +47,22 @@ object URL                                                             {
 
     } yield url
 
-  def asString(url: URL): String = url.kind match {
-    case Location.Relative                     => url.path.asString
-    case Location.Absolute(scheme, host, port) =>
-      port match {
-        case 80   => s"${scheme.asString}://${host}${url.path.asString}"
-        case port => s"${scheme.asString}://${host}:${port}${url.path.asString}"
+  def asString(url: URL): String = {
+    def withQuery(u: String): String =
+      if (url.query.nonEmpty) {
+        s"$u?${url.query}"
+      } else {
+        u
       }
+
+    url.kind match {
+      case Location.Relative                     => withQuery(url.path.asString)
+      case Location.Absolute(scheme, host, port) =>
+        (scheme, port) match {
```

Comment:
Tuple Boxing will incur additional allocations and might affect perf.

Reply:
It's producing a string. Not typically in a hot path. Did it materially affect benchmarks?

The main point was to convey that the existing logic had a bug. HTTPS and port 80 would have omitted the port but the default port for https is 443.